src/QtBase/FileManager.cpp

```diff
@@ -179,8 +191,26 @@ void FileManager::OpenClassifier(ELocation location, const char* identifier, con
 		if (CloseWithPrompt(activeClassifier->GetUuid()) == false)
 			return;	// user aborted
 
+	if (mIsInFileSaving)
+	{
+		//wait for backend reply
+		QTimer timer;
+		timer.setSingleShot(true);
+		QEventLoop loop;
+		connect(this, &FileManager::writeToBackendFinished, &loop, &QEventLoop::quit);
+		connect(&timer, &QTimer::timeout, &loop, &QEventLoop::quit);
+		timer.start(10000);
+		loop.exec();
```

Comment:
This "manuall syncing" looks really a bit hacky to me...

Wouldn't this `exec()` block the calling thread until either a response has been received, or 10s have been elapsed? I think blocking the calling thread (it's even the mainthread which handles ui and input here) isn't a good idea. Also running Thread Event Loops inside a Thread Event Loop (which this likely is), also seems a bad approach to me?

For testing purposes, can you please modify the function that makes the request: Add something like a sleep 20 to simulate a connection loss/issues and see how all this code behaves, IF it actually runs into the 10s timeout?

Reply:
Ok. Not an easy topic. I will try to investigate deeper into Qt Looper and this specific issue here.
But let's try to sync on a common understanding first:

**Some Minimal Example Application Looper**
```
void update() {
  doSomething()
  doCritical()
}
void main() {
  while(running) {
    update()
    messagePump()
    sleep()
  }
}
```

Simple solutions usually try to achieve a fixed update-rate, e.g. 60 Hz/FPS. More advanced solutions would maintain a task schedule and try to execute the tasks accurately timed, sleeping whenever there is nothing to do.

**Blocking**
Now, let's assume, doCritical() should be stopped from executing for a while.
From what I understand, you have chosen to achieve it by doing something similar to this:

```
void update() {
  doSomething()
  while(waiting) {
    messagePump()
    sleep()
  }
  doCriticial()
}
void main() {
  while(running) {
    update()
    messagePump()
    sleep()
  }
}
```

I believe you if you say that Qt somehow may be able to manage a Looper inside a Looper (also inside a Looper inside a Looper inside a Looper...?). But creating and running another looper inside a looper is still bad design.
Instead, please try to find a solution which avoids blocking the main looper from continuing its regular execution:

```
void update() {
  doSomething()
  if (!waiting) {
    doCriticial()
  }
}
void main() {
  while(running) {
    update()
    messagePump()
    sleep()
  }
}
```

I know it looks a lot easier in these minimal examples, but the basic concept stays the same...